pkg/placement/placement-resolver.go

```diff
@@ -0,0 +1,225 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// A PlacementResolver holds a collection of placement resolutions.
+// The collection is indexed by placementKey string, the resolver does not
+// care what the strings are, but a string should be a placement's unique
```

Comment:
FYI: Saying that a string "should be" something _here_ suggests that it matters to the PlacementResolver. I think that is false, the comment about strings being a unique identifier for a Placement belongs higher in the stack.